Adding Tool Error tracking

kellrott opened the pull request:
Traditionally errors related tool xml formatting or issues that occurred during processing the command line (the 'failure preparing job' message) have required tool developers to dig into server logs to find out what is causing the problem. 
This PR adds a system to track those errors and view them via the API and a page on the admin panel.

kellrott committed:
Adding system to log tool loading/preperation exceptions so that they can be viewed in an admin dialog

kellrott committed:
Adding @web.require_admin to /api/error/tools

nsoranzo commented on lib/galaxy/tools/__init__.py:
```diff
@@ -80,6 +82,24 @@
+        self.error_stack.insert(0, {
+            "file" : file,
+            "time" : str(datetime.now()),
+            "section" : section,
```
What about "stage" or "phase" instead of "section"?

nsoranzo commented on templates/admin/tool_errors.mako:
```diff
@@ -0,0 +1,27 @@
+<table class="manage-table colored" border="0" cellspacing="0" cellpadding="0" width="100%">
+<tr class="header">
+<td>Time</td>
+<td>System</td>
```
As above, what about "stage" or "phase"?

kellrott committed:
Removing /api/errors for the moment (the admin control panel doesn't use the API, so its not needed)

kellrott committed:
Clarifing naming to use 'Phase'

kellrott commented:
I changed the wording to use 'Phase' to describe when an error was occurring. 
Also I removed the API component, based on offline comments from @jmchilton

nsoranzo commented:
:+1:

dannon commented:
+1
